webapp/src/components/CreateEstatePage/EstateSelect/EstateSelect.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Container } from 'semantic-ui-react'
+
+import ParcelPreview from 'components/ParcelPreview'
+import EstateSelectActions from './EstateSelectActions'
+import Parcel from 'components/Parcel'
+import { t } from 'modules/translation/utils'
+import { isOwner } from 'modules/parcels/utils'
+import { coordsType } from 'components/types'
+
+import './EstateSelect.css'
+
+export default class EstateSelect extends React.PureComponent {
+  static propTypes = {
+    x: PropTypes.number.isRequired,
+    y: PropTypes.number.isRequired,
+    error: PropTypes.string,
+    onCancel: PropTypes.func.isRequired,
+    onContinue: PropTypes.func.isRequired,
+    onChange: PropTypes.func.isRequired,
+    value: PropTypes.arrayOf(coordsType).isRequired
+  }
+
+  isNeighbour = (x, y) => coords => {
+    return (
+      (coords.x === x && (coords.y + 1 === y || coords.y - 1 === y)) ||
+      (coords.y === y && (coords.x + 1 === x || coords.x - 1 === x))
+    )
+  }
+
+  hasNeighbour = (x, y) => {
+    const { value: parcels } = this.props
+    return parcels.some(this.isNeighbour(x, y))
+  }
+
+  getNeighbours = (x, y, parcels = this.props.parcels) => {
+    return parcels.filter(this.isNeighbour(x, y))
+  }
+
+  areConnected = (parcels, alreadyTraveled = [], remaining = [...parcels]) => {
+    if (alreadyTraveled.length === parcels.length) {
+      return true
+    }
+
+    if (remaining.length === 0) {
+      return false
+    }
+
+    let actual = remaining.pop()
+
+    const neighbours = this.getNeighbours(actual.x, actual.y, parcels).filter(
+      coords => {
+        return (
+          parcels.some(
+            coords2 => coords.x === coords2.x && coords.y === coords2.y
+          ) &&
+          !alreadyTraveled.some(
+            coords2 => coords.x === coords2.x && coords.y === coords2.y
+          )
+        )
+      }
+    )
```

Comment:
This is a bit confusing to me, calling reading the code I don't see what `neighbours` is at first glance. Can we maybe name this entire method?

```javascript
this.getNeighbours(actual.x, actual.y, parcels).filter(
      coords => {
        return (
          parcels.some(
            coords2 => coords.x === coords2.x && coords.y === coords2.y
          ) &&
          !alreadyTraveled.some(
            coords2 => coords.x === coords2.x && coords.y === coords2.y
          )
        )
      }
    )
```

Reply:
I call `neighbour` to a parcel that the user also own and at the same time is next to another parcel, do you have any alternative to propose?

Reply:
contiguous